### PR TITLE
ci: implement GitHub Actions CI

### PR DESCRIPTION
**Pull request by icyphox**
## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

## Further comments

This PR implements CI in GitHub Actions, transitioning away from CircleCI. There are related changes in [ory/ci](https://github.com/ory/ci) as well. Below are the required secrets for the workflow to function:

- `MAILCHIMP_API_KEY`
- `SLACK_WEBHOOK_URL` — configure this in [Incoming Webhooks](https://api.slack.com/incoming-webhooks).
- `GORELEASER_KEY` — key for GoReleaser Pro.
- `NPM_TOKEN`
- `SSH_KEY`
- `PAT`

This also requires the creation of an [environment](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) called `production`. This environment needs to be configured with [required reviewers](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#required-reviewers). This ensures that the `newsletter-send` job is protected behind manual approval. Also see: https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments

NOTE: This PR depends on https://github.com/ory/ci/pull/59 for the `releaser` and `newsletter` actions.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  release:
+    name: Generate release
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_type == 'tag' }}
+    needs:
+      - test
+    steps:
+      - uses: ory/ci/releaser@master
+        with:
+          token: ${{ secrets.PAT }}
```
Comment (aeneasr) [inline .github/workflows/ci.yaml:61]: Can we scope the permissions needed for this token?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+          go-version: '^1.17'
+      - run: |
+          mkdir -p docs/docs
+          make docs/cli
+          echo 'docsdir='"./docs" >> $GITHUB_ENV
+      - uses: actions/checkout@v2
+        with:
+          repository: ory/docs
+          path: ${{ env.docsdir }}
+          token: ${{ secrets.PAT }}
```
Comment (aeneasr) [inline .github/workflows/ci.yaml:40]: Can we scope the permissions needed for this token?

**Review by aeneasr — COMMENTED** [review]
Yo, this looks sweet!

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+      - env:
+          GITHUB_TOKEN: ${{ secrets.PAT }}
```
[the remark this answers is not in the document]
Comment (icyphox) [reply .github/workflows/ci.yaml:42]: I think we can, but I'm not sure — `act` was complaining. As for scopes, repos and org:read should be enough.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (icyphox) [reply .github/workflows/ci.yaml:61]: Same as above.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (icyphox) [reply .github/workflows/ci.yaml:40]: Same as above.

Comment (aeneasr) [reply .github/workflows/ci.yaml:40]: Thanks! To ensure we do not have any collisions, could you rename this secret to `ORY_BOT_PAT`? :)

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    needs:
+      - release
+    steps:
+      - uses: ory/ci/newsletter@master
+        with:
+          mailchimp_list_id: 058a056618
+          mailchmip_segment_id: 11398953
+          mailchimp_api_key: ${{ secrets.MAILCHIMP_API_KEY }}
+          draft: 'true'
+          ssh_key: ${{ secrets.SSH_KEY }}
```
Comment (aeneasr) [inline .github/workflows/ci.yaml:97]: and for this also
```suggestion
          ssh_key: ${{ secrets.ORY_BOT_SSH_KEY }}
```

**Pull request merged**